fedimint-client-legacy/src/mint/mod.rs

```diff
@@ -383,13 +384,87 @@ impl MintClient {
         NoteIssuanceRequest::new(ctx, secret)
     }
 
+    pub async fn summary(&self) -> TieredSummary {
+        self.start_dbtx()
+            .await
+            .find_by_prefix(&NoteKeyPrefix)
+            .await
+            .fold(
+                TieredSummary::default(),
+                |mut acc, (key, _note)| async move {
+                    acc.inc(key.amount, 1);
+                    acc
+                },
+            )
+            .await
+    }
+
+    /// Select notes with total amount of *at least* `amount`. If more than
+    /// requested amount of notes are returned it was because exact change
+    /// couldn't be made, and the next smallest amount will be returned.
+    ///
+    /// The caller can request change from the federation.
     pub async fn select_notes(&self, amount: Amount) -> Result<TieredMulti<SpendableNote>> {
-        let notes = self.notes().await;
-        let selected_notes = notes.select_notes(amount).ok_or_else(|| {
-            MintClientError::InsufficientBalance(amount, TieredMulti::total_amount(&notes))
-        })?;
+        let mut t = self.start_dbtx().await;
```

Comment:
nit: I prefer `dbtx` for var names since fedimint has multiple definitions of a "transaction" and its easy to get confused (likely not in this function but good to be consistent)